armi/reactor/blocks.py

```diff
@@ -1580,61 +1580,85 @@ def setPinPowers(
         numPins,
         imax,
         jmax,
-        gamma=False,
         removeSixCornerPins=False,
         powerKeySuffix="",
     ):
         """
-        Updates the pin powers of this block for the current rotation.
+        Updates the pin linear power densities of this block for the current rotation.
+        The linear densities are represented by the *linPowByPin* parameter.
 
         Parameters
         ----------
-        powers : list of floats
-            The block-level pin linear power densities. pinPowers[i] represents the average linear
-            power density of pin i.
-            Power units are Watts/cm (Watts produced per cm of pin length).
-            The "ARMI pin ordering" is used, which is counter-clockwise from 3 o'clock.
+        powers : list of floats, required
+            The block-level pin linear power densities. powers[i] represents the average
+            linear power density of pin i. The units of linear power density is watts/cm
+            (i.e., watts produced per cm of pin length). The "ARMI pin ordering" must be
+            be used, which is counter-clockwise from 3 o'clock.
 
-        Notes
-        -----
-        This handles rotations using the pinLocation parameters.
+        numPins : int, required
+            Number of pins in the block. This parameter should probably be removed because it
+            can be derived from the block.
 
-        This sets:
+        imax: int, required
+            Number of pin rings. This parameter should probably be removed because it
+            can be derived from the block.
```

Comment:
imax and jmax can be completely removed and derived using grids.py now.

See: `_getPinCoordinatesHex` for an example of how the hexagon locations for i, j are determined based on the pin pitch and the number of pinned components within the block. This would be better to use the grids system since it is newer than this code now.